File: 769.最多能完成排序的块.py
#
# @lc app=leetcode.cn id=769 lang=python3
#
# [769] 最多能完成排序的块
#
from typing import *
# @lc code=start
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        ret = 0
        cache_a, cache_b = 0, 0
        for i, item in enumerate(arr):
            cache_a += i
            cache_b += item
            if cache_a == cache_b:
                ret += 1
        return ret
# @lc code=end

File: 769.最多能完成排序的块.rs
/*
 * @lc app=leetcode.cn id=769 lang=rust
 *
 * [769] 最多能完成排序的块
 */

// @lc code=start
impl Solution {
    pub fn max_chunks_to_sorted(arr: Vec<i32>) -> i32 {
        let mut ret = 0;
        arr.into_iter().enumerate().fold((0usize, 0), |mut c, item_pair|{
            c.0 += item_pair.0;
            c.1 += item_pair.1;
            if c.0 as i32 == c.1{
                ret += 1;
            }
            c
        });
        ret
    }
}
// @lc code=end

